addons/hr_contract/models/hr_contract.py

```diff
@@ -258,6 +258,8 @@ def _safe_write_for_cron(self, vals, from_cron=False):
 
     def _assign_open_contract(self):
         for contract in self:
+            if contract.employee_id.contract_id.state == 'open':
+                continue
```

Comment:
This does not work if the contract in the future was created and set to running first